### ISSUE TITLE
Crash on setting tags or downloaded status

### ISSUE THREAD

**Issue by manalive** [rmcrackan/Libation#841]
**Describe the bug**
The behavior appears to be the same as #820.

When attempting to modify tags, or set status to Downloaded, Libation crashes.
Downloads appear to stall at 99% for some time; then there is a crash. 

Crash happens whether setting tags on individual books or using the _Visible Books_ menu.

**Platform**

Edition	Windows 10 Enterprise
Version	22H2
OS build	19045.3930
Experience	Windows Feature Experience Pack 1000.19053.1000.0

Libation v11.3.4.1 Chardonnay

**Log Files**
I cleared log files and the temp directory. then opened the application (logging set to verbose), searched for a book and tried to set a tag value. The attached file is the resulting log, which shows some errors during launch but simply cuts off before the error. 
[_Log202402.zip](https://github.com/rmcrackan/Libation/files/14201867/_Log202402.zip)

Like one of the users on #820 I ran debug on the app in VS2022 and it shows a StackOverflow exception being thrown. (Not being a dev, I don't know how to interpret that or proceed.)

**Comment by rmcrackan** [comment]
This is a weird one. Could you back up your database file `LibationContext.db`, delete the original file, restart Libation and do a new scan? If that works, then maybe the file was corrupted. If this works, let's see if I can get into the old file enough to retrieve download statuses and tags.

**Comment by manalive** [comment]
Thank you for your reply @rmcrackan!

I've since performed the following steps:

1. Backed up LibationContext.db, and removed it. 
2. Launched Libation, added account, scanned, reapplied all my normal settings

**Same behavior on attempting to set a tag value.**

1. Removed the entire Libation folder from my user directory (and deleted the log file from AppData\Local\Temp)
2. Redownloaded Libation v11.3.4.1 Chardonnay app files and ran it
3. Added account, scanned, reapplied all my normal settings

**Same behavior on attempting to set a tag value.**

1. Downloaded Libation v11.1.0.1 application files and launched it (did not make any changes to db or other file(

**Same behavior on attempting to set a tag value.**

1. Deleted app folder again, ran v11.1.0.1 as new user
2. Added account, scanned (did not change settings)

**Download successful, tag setting successful**

1. Reapplied usual settings

**Download successful, tag setting successful**

1. Closed Libation and ran v 11.3.4.1

**Crashes when setting tag**

1. Closed Libation and ran v11.1.0.1

**Crashes when setting tag**

_So this SEEMS to be an issue with 11.3.4.1 (and my system or settings) that corrupts the database._

**Comment by rmcrackan** [comment]
Wow -- thank you for this exhaustive testing and reporting. Apparently I just can't seem to get 11.3 right. I can't imagine what's corrupting the files.

**Comment by manalive** [comment]
> Wow -- thank you for this exhaustive testing and reporting. Apparently I just can't seem to get 11.3 right. I can't imagine what's corrupting the files.

Major change seems to be .net8 ?

If you come up with any ideas, I don't have any problem testing a build. @rmcrackan

**Comment by rmcrackan** [comment]
Yeah, it was evidently something to do with the .net8 upgrade. Everything is working fine on my pc. I upgraded a boatload of dependencies to .net8 so I'd think they'd be fine also. I'm clearly missing something though. My latest attempt was 11.3.4 which is the one you're having issues with. I'm not sure what else to change.

**Comment by rmcrackan** [comment]
Please try the latest version: https://github.com/rmcrackan/Libation/releases/latest
